### PR TITLE
Fixes tab pages after setting change

### LINKED ISSUE #7806: Blank tabs bar appears after increasing the number of tabs per tab set
## Test plan

https://github.com/brave/browser-laptop/pull/9773#issue-239452018

----

**Describe the issue you encountered:**
the tabs bar becomes blank after increasing the number of tabs per tab set under a condition. I believe the same kind of issue had been fixed once and then regressed, recently or not.

- Platform (Win7, 8, 10? macOS? Linux distro?):

- Brave Version (revision SHA): 64b381a

- Steps to reproduce:
    1. Open about:preferences#tabs
    2. Set the number of tabs per tab set to 6
    3. Open new tabs to create a new tab set, which has just **2** tabs
    4. Increase the number to 8
    
- Actual result: the blank tabs bar which has zero tabs appears
screenshot 2017-03-21 23 07 46

- Expected result: the 2nd tab set should be closed

- Any related issues:

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #7806

Auditors: @bsclifton @cezaraugusto

Test Plan:
- Open about:preferences#tabs
- Set the number of tabs per tab set to 6
- Open new tabs to create a new tab set, which has just 2 tabs
- Increase the number to 8

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: test/tab-components/tabPagesTest.js [hunk 1]
```diff
@@ -90,10 +91,20 @@ describe('tab pages', function () {
...
-        const defaultTabsPerPage = appConfig.defaultSettings[settings.TABS_PER_PAGE]
+        const newValue = 6
+        const tabs = appConfig.defaultSettings[settings.TABS_PER_PAGE]
+        const numberOfPages = Math.ceil(tabs / newValue)
+        const numberOfTabs = tabs - (Math.floor(tabs / newValue) * newValue)
         yield this.app.client
-          .changeSetting(settings.TABS_PER_PAGE, 1)
-          .waitForElementCount(tabPage, defaultTabsPerPage)
+          .waitForBrowserWindow()
+          .loadUrl('about:preferences')
```
Comment (NejcZdovc) [inline test/tab-components/tabPagesTest.js]: This is not loaded in the right context, but I am not sure how to switch to the right one

Comment (bbondy) [inline test/tab-components/tabPagesTest.js]: tabByIndex?

Comment (NejcZdovc) [inline test/tab-components/tabPagesTest.js]: Problem is that is not working for me. Most of the times we use if with 0, maybe 1. But my use case is that I need to focus last active tab, which is this case 19 (we have 20 tabs). But if do `.tabByIndex(19)` second tab is selected.

Comment (NejcZdovc) [inline test/tab-components/tabPagesTest.js]: Just pushed new version with `tabByIndex(19)` so you can try it out. I am thinking that something is not ok with `handles`?

Comment (NejcZdovc) [inline test/tab-components/tabPagesTest.js]: test fixed based on @cezaraugusto findings

**Comment by NejcZdovc** [comment]
@cezaraugusto PR is now ready for a review, test was fixed based on your discovery. Thank you

**Comment by NejcZdovc** [comment]
@luixxiul thank you for your test

**Pull request merged**